From 1f26a8d5d7325d7901910f1152acf34aeadcc921 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 22 Feb 2022 09:51:29 +0800
Subject: [PATCH 53/58] HACK: compositor: Re-build view list when framing an
 unassigned surface

To avoid skipping repaint for unassigned surface and delaying the frame
callback.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/compositor.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libweston/compositor.c b/libweston/compositor.c
index 0188e2d..bc018f0 100644
--- a/libweston/compositor.c
+++ b/libweston/compositor.c
@@ -3603,6 +3603,9 @@ surface_frame(struct wl_client *client,
 
 	wl_list_insert(surface->pending.frame_callback_list.prev,
 		       wl_resource_get_link(cb));
+
+	if (!surface->output_mask)
+		weston_compositor_build_view_list(surface->compositor, NULL);
 }
 
 static void
-- 
2.20.1

