From baff016bed15e9f1735fd66c5d100f0e88d3e7c6 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 22 Feb 2022 09:51:29 +0800
Subject: [PATCH 55/75] HACK: compositor: Re-build view list when committing an
 unassigned surface

To avoid skipping repaint for unassigned surface.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/compositor.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libweston/compositor.c b/libweston/compositor.c
index 511d1abe3..1a8ef1297 100644
--- a/libweston/compositor.c
+++ b/libweston/compositor.c
@@ -4034,6 +4034,10 @@ weston_surface_commit(struct weston_surface *surface)
 
 	weston_surface_commit_subsurface_order(surface);
 
+	/* HACK: Assign outputs */
+	if (!surface->output_mask)
+		weston_compositor_build_view_list(surface->compositor, NULL);
+
 	weston_surface_schedule_repaint(surface);
 }
 
-- 
2.20.1

