From 9748ddc88a07f880abf87e748eb9084be7377a5a Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Thu, 20 Jun 2019 12:43:20 +0800
Subject: [PATCH 05/46] kmssink: Request window handle

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 sys/kms/gstkmssink.c | 13 +++++++++++++
 sys/kms/gstkmssink.h |  2 ++
 2 files changed, 15 insertions(+)

diff --git a/sys/kms/gstkmssink.c b/sys/kms/gstkmssink.c
index a9181465..0da40562 100644
--- a/sys/kms/gstkmssink.c
+++ b/sys/kms/gstkmssink.c
@@ -165,11 +165,22 @@ gst_kms_sink_expose (GstVideoOverlay * overlay)
   gst_kms_sink_show_frame (GST_VIDEO_SINK (self), NULL);
 }
 
+static void
+gst_kms_sink_set_window_handle (GstVideoOverlay * overlay, guintptr handle)
+{
+  GstKMSSink *self = GST_KMS_SINK (overlay);
+
+  g_return_if_fail (self != NULL);
+
+  self->window_handle = handle;
+}
+
 static void
 gst_kms_sink_video_overlay_init (GstVideoOverlayInterface * iface)
 {
   iface->expose = gst_kms_sink_expose;
   iface->set_render_rectangle = gst_kms_sink_set_render_rectangle;
+  iface->set_window_handle = gst_kms_sink_set_window_handle;
 }
 
 static int
@@ -1157,6 +1168,8 @@ gst_kms_sink_set_caps (GstBaseSink * bsink, GstCaps * caps)
   }
   GST_OBJECT_UNLOCK (self);
 
+  gst_video_overlay_prepare_window_handle (GST_VIDEO_OVERLAY (bsink));
+
   GST_DEBUG_OBJECT (self, "negotiated caps = %" GST_PTR_FORMAT, caps);
 
   return TRUE;
diff --git a/sys/kms/gstkmssink.h b/sys/kms/gstkmssink.h
index 428c057e..0fccfa27 100644
--- a/sys/kms/gstkmssink.h
+++ b/sys/kms/gstkmssink.h
@@ -93,6 +93,8 @@ struct _GstKMSSink {
   /* reconfigure info if driver doesn't scale */
   GstVideoRectangle pending_rect;
   gboolean reconfigure;
+
+  guintptr window_handle;
 };
 
 struct _GstKMSSinkClass {
-- 
2.20.1

