From e87c430bb24078e127b628c3735cb855af1a5a2a Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Wed, 25 Nov 2020 10:51:12 +0800
Subject: [PATCH 13/49] waylandsink: Drop frame when window not ready

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 ext/wayland/gstwaylandsink.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 7e146fe8..a4de9ab2 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -926,10 +926,15 @@ render:
 
 no_window_size:
   {
+    /* HACK: Drop frame when window not ready */
+#if 0
     GST_ELEMENT_ERROR (self, RESOURCE, WRITE,
         ("Window has no size set"),
         ("Make sure you set the size after calling set_window_handle"));
     ret = GST_FLOW_ERROR;
+#else
+    GST_WARNING_OBJECT (self, "Window has no size set");
+#endif
     goto done;
   }
 no_buffer:
-- 
2.20.1

