From cb6a2ad8c6dab0e026dee60fbdd29e04f234ed5e Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 26 Mar 2019 17:33:37 +0800
Subject: [PATCH 22/45] HACK: kmssink: Open drm devnode directly

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 sys/kms/gstkmssink.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sys/kms/gstkmssink.c b/sys/kms/gstkmssink.c
index b25e599..de29e46 100644
--- a/sys/kms/gstkmssink.c
+++ b/sys/kms/gstkmssink.c
@@ -57,6 +57,7 @@
 #include <xf86drmMode.h>
 #include <drm_fourcc.h>
 #include <string.h>
+#include <fcntl.h>
 
 #include "gstkmssink.h"
 #include "gstkmsutils.h"
@@ -1123,6 +1124,9 @@ gst_kms_sink_start (GstBaseSink * bsink)
       self->fd = kms_open (&self->devname);
   }
 
+  if (self->fd < 0)
+    self->fd = open ("/dev/dri/card0", O_RDWR);
+
   if (self->fd < 0)
     goto open_failed;
 
-- 
2.20.1

