From bc9cce551f86fd165cee388adb74c97a3fd52579 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Thu, 20 Jun 2019 10:51:32 +0800
Subject: [PATCH] autodetect: Add dmabuf feature in video caps

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 gst/autodetect/gstautodetect.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/gst/autodetect/gstautodetect.c b/gst/autodetect/gstautodetect.c
index 3e30241..3971099 100644
--- a/gst/autodetect/gstautodetect.c
+++ b/gst/autodetect/gstautodetect.c
@@ -36,6 +36,10 @@ GST_DEBUG_CATEGORY (autodetect_debug);
 
 #define DEFAULT_SYNC                TRUE
 
+#ifndef GST_CAPS_FEATURE_MEMORY_DMABUF
+#define GST_CAPS_FEATURE_MEMORY_DMABUF "memory:DMABuf"
+#endif
+
 /* Properties */
 enum
 {
@@ -169,7 +173,7 @@ gst_auto_detect_reset (GstAutoDetect * self)
 }
 
 static GstStaticCaps raw_audio_caps = GST_STATIC_CAPS ("audio/x-raw");
-static GstStaticCaps raw_video_caps = GST_STATIC_CAPS ("video/x-raw");
+static GstStaticCaps raw_video_caps = GST_STATIC_CAPS ("video/x-raw ; video/x-raw(" GST_CAPS_FEATURE_MEMORY_DMABUF ")");
 
 static void
 gst_auto_detect_init (GstAutoDetect * self)
-- 
2.11.0

