From 15f1c32d3e9e48043795682f63e4bd59dc4d45b2 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 28 Sep 2020 20:02:00 +0800
Subject: [PATCH 09/16] media: gpu: v4l2: Fix compile error when ozone not
 enabled

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 media/gpu/v4l2/BUILD.gn | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/media/gpu/v4l2/BUILD.gn b/media/gpu/v4l2/BUILD.gn
index 5f6eb8d25..a78620c5a 100644
--- a/media/gpu/v4l2/BUILD.gn
+++ b/media/gpu/v4l2/BUILD.gn
@@ -144,9 +144,12 @@ source_set("v4l2") {
     "//third_party/libgav1:libgav1_parser",
     "//third_party/libyuv",
     "//ui/gfx/geometry",
-    "//ui/ozone",
   ]
 
+  if (use_ozone) {
+    deps += [ "//ui/ozone" ]
+  }
+
   if (is_chromeos_ash) {
     sources += [
       "v4l2_jpeg_encode_accelerator.cc",
-- 
2.38.5

