From 66fad78ab7f1237848ab3dc1484f8bf5e63791a5 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 17 Nov 2020 11:51:40 +0800
Subject: [PATCH 12/49] gstjpegparse: Allow parsebin to use it for autoplugging

Increase rank.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 gst/jpegformat/gstjpegparse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/jpegformat/gstjpegparse.c b/gst/jpegformat/gstjpegparse.c
index 6be24a28..3c549030 100644
--- a/gst/jpegformat/gstjpegparse.c
+++ b/gst/jpegformat/gstjpegparse.c
@@ -107,7 +107,7 @@ static gboolean gst_jpeg_parse_stop (GstBaseParse * parse);
 
 #define gst_jpeg_parse_parent_class parent_class
 G_DEFINE_TYPE (GstJpegParse, gst_jpeg_parse, GST_TYPE_BASE_PARSE);
-GST_ELEMENT_REGISTER_DEFINE (jpegparse, "jpegparse", GST_RANK_NONE,
+GST_ELEMENT_REGISTER_DEFINE (jpegparse, "jpegparse", GST_RANK_SECONDARY,
     GST_TYPE_JPEG_PARSE);
 
 enum GstJPEGColorspace
-- 
2.20.1

