From bc9ce5407d5d7b4de0c7dd7a9a98f3aa86847420 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 13 Feb 2023 15:30:55 +0800
Subject: [PATCH 02/17] HACK: media: gpu: v4l2: Enable V4L2 VEA

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 media/gpu/v4l2/BUILD.gn | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/media/gpu/v4l2/BUILD.gn b/media/gpu/v4l2/BUILD.gn
index 5aa6d8c76..6c754a3d8 100644
--- a/media/gpu/v4l2/BUILD.gn
+++ b/media/gpu/v4l2/BUILD.gn
@@ -89,7 +89,12 @@ source_set("v4l2") {
       # AV1 delegate depends on header files only in ChromeOS SDK
       "v4l2_video_decoder_delegate_av1.cc",
       "v4l2_video_decoder_delegate_av1.h",
+    ]
+  }
 
+  # HACK: Enable V4L2 VEA for v4l-rkmpp.
+  if (true) {
+    sources += [
       # TODO(crbug.com/901264): Encoders use hack for passing offset
       # within a DMA-buf, which is not supported upstream.
       "v4l2_video_encode_accelerator.cc",
-- 
2.38.5

