From 523b171f8686b9b36da631543651c27875772fc7 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 28 Sep 2020 20:02:00 +0800
Subject: [PATCH 09/17] media: gpu: v4l2: Fix compile error when ozone not
 enabled

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 media/gpu/v4l2/BUILD.gn | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/media/gpu/v4l2/BUILD.gn b/media/gpu/v4l2/BUILD.gn
index 24700df35..0d93063c0 100644
--- a/media/gpu/v4l2/BUILD.gn
+++ b/media/gpu/v4l2/BUILD.gn
@@ -133,9 +133,12 @@ source_set("v4l2") {
     "//media/parsers",
     "//third_party/libyuv",
     "//ui/gfx/geometry",
-    "//ui/ozone",
   ]
 
+  if (use_ozone) {
+    deps += [ "//ui/ozone" ]
+  }
+
   if (is_chromeos_ash) {
     sources += [
       "v4l2_jpeg_encode_accelerator.cc",
-- 
2.38.5

