From 514b2447c509e018ce5ba60117999aaec569f2ff Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Fri, 5 Aug 2022 11:27:54 +0800
Subject: [PATCH 75/93] gl-renderer: Clear RGB when alpha is zero in fast path

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/renderer-gl/fragment.glsl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libweston/renderer-gl/fragment.glsl b/libweston/renderer-gl/fragment.glsl
index a950578..e294afd 100644
--- a/libweston/renderer-gl/fragment.glsl
+++ b/libweston/renderer-gl/fragment.glsl
@@ -288,6 +288,8 @@ main()
 
 	if (c_need_color_pipeline)
 		color = color_pipeline(color); /* Produces straight alpha */
+	else if (color.a == 0.0)
+		color.rgb = vec3(0, 0, 0);
 
 	/* Ensure pre-multiplied for blending */
 	if (!c_input_is_premult || c_need_color_pipeline)
-- 
2.20.1

