From 512ab9ef1d7d3ae2da6111555126db457894c99f Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Wed, 18 Dec 2019 09:21:34 +0800
Subject: [PATCH] media: gpu: v4l2: Fix component compile error

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 media/gpu/v4l2/BUILD.gn | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/media/gpu/v4l2/BUILD.gn b/media/gpu/v4l2/BUILD.gn
index 4bdc884..09c97b7 100644
--- a/media/gpu/v4l2/BUILD.gn
+++ b/media/gpu/v4l2/BUILD.gn
@@ -34,8 +34,6 @@ source_set("v4l2") {
     "v4l2_image_processor.h",
     "v4l2_vda_helpers.cc",
     "v4l2_vda_helpers.h",
-    "v4l2_jpeg_encode_accelerator.cc",
-    "v4l2_jpeg_encode_accelerator.h",
     "v4l2_video_decode_accelerator.cc",
     "v4l2_video_decode_accelerator.h",
     "v4l2_video_encode_accelerator.cc",
@@ -104,6 +102,8 @@ source_set("v4l2") {
 
   if (is_chromeos) {
     sources += [
+      "v4l2_jpeg_encode_accelerator.cc",
+      "v4l2_jpeg_encode_accelerator.h",
       "v4l2_mjpeg_decode_accelerator.cc",
       "v4l2_mjpeg_decode_accelerator.h",
     ]
-- 
2.11.0

